### DISCUSSION TITLE
Have to press <Tab>, then <S-Tab> to get the first candidate

### DISCUSSION THREAD

**Discussion by archibate** [neoclide/coc.nvim#4088] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I press Tab, it gives the **second candidate** instead of the first.
So in order to trigger completion for the first candidate, I have to **press Tab then S-Tab**, which is annoying.

Input `inorem`...
![image](https://user-images.githubusercontent.com/20640597/186209645-81c5f0e3-58ae-452f-83da-e3799fb86be5.png)
Press `<Tab>`...
![image](https://user-images.githubusercontent.com/20640597/186209697-d1d39698-63c6-4b9e-807a-26181673c6ac.png)
Press `<S-Tab>`...
![image](https://user-images.githubusercontent.com/20640597/186209921-ed0027c9-da89-4d8c-82d1-4933236653e1.png)

**Describe the solution you'd like**
I guess this is because the new version of coc.nvim selects the first candidate at the first place.
However, the first candidate hasn't been triggered at all.
And when I press Tab, it performs `coc#pum#next(1)` and goes to the second candidate and triggered completion via `coc#pum#refresh()`.
An old version coc.nvim selects no candidate at the first place (say 0th candidate), and when press Tab it normally completes the first candidate (I like this).
So may I have an option to get back the behavior as old version does, e.g. `let g:coc_select_0th_candidate = 1` to select no candidate at the first place, so that I can press `<Tab>` once to complete the first candidate?

**Describe alternatives you've considered**
Use the `<CR>` mapping mentioned in README.md may also works, so that I will use `<CR>` for the first candidate and `<Tab>` for the second candidate. But I have mapped `<CR>` for other plugins, and I still prefer `<Tab>` in muscle memory anyway :rofl: ...

**Additional context**
Here is what I used in `.vimrc` (copied from README.md of this repo):
```
" Use tab for trigger completion with characters ahead and navigate.
" NOTE: Use command ':verbose imap <tab>' to make sure tab is not mapped by
" other plugin before putting this into your config.
inoremap <silent><expr> <TAB>
      \ coc#pum#visible() ? coc#pum#next(1) :
      \ <SID>check_back_space() ? "\<TAB>" :
      \ coc#refresh()
inoremap <expr><S-TAB> coc#pum#visible() ? coc#pum#prev(1) : "\<C-h>"

function! s:check_back_space() abort
  let col = col('.') - 1
  return !col || getline('.')[col - 1]  =~# '\s'
endfunction
```

**Comment by koreanddinghwan** [comment]
I am also troubled by this problem these days.
Various problems in app have arisen since the recent plug-in update.

**Comment by chemzqm** [comment, marked as answer]
> " NOTE: There's always complete item selected by default, you may want to enable no select by `"suggest.noselect": true` in your configuration file.

From Readme

**Comment by chemzqm** [comment]
Or use `coc#pum#insert()` to insert word of current complete item.
This is my key-mapping for `<tab>`
``` vim
    inoremap <silent><expr> <TAB>
          \ coc#pum#visible() ? coc#pum#insert():
          \ <SID>check_back_space() ? "\<Tab>" :
          \ coc#refresh()
    inoremap <expr><S-TAB> coc#pum#visible() ? coc#pum#prev(1) : "\<C-h>"
```
From coc.nvim 0.0.82, the default behavior changed to match the behavior of VSCode, while also makes preselect and selection features just work

**Reply by albingroen** [reply]
I actually prefer the old behavior. How would I revert to that?

**Reply by albingroen** [reply]
This was still the answer apparently

```json
{
  "suggest": {
    "noselect": true
  }
}
```

**Reply by chemzqm** [reply]
You should use `"suggest.noselect": true`

**Comment by archibate** [comment]
Thanks! `"suggest.noselect": true` just works for me! Using `coc#pum#insert()` actually make `<Tab>` being same as `<CR>` (which becomes asymmetric with `<S-Tab>`), and we need to use arrow `<Down>` instead for next candidate after doing so (like VSCode does).

**Reply by chemzqm** [reply]
You can use `<C-n>` and `<C-p>` to navigate the list.

**Reply by xixa** [reply]
`"suggest.noselect": true` worked for `<CR>` but when I use `<C-n>` or `<C-p>` I get the following:
```
Error detected while processing function <lambda>54[1]..<SNR>171_navigate[5]..<SNR>171_select_by_index[2]..<SNR>171_set_cursor[21]..<SNR>171_select_line[8]..coc#dialog#place_sign:
line    3:
E155: Unknown sign: CocCurrentLine
```
Any suggestion?

**Reply by chemzqm** [reply]
No idea, try install latest coc.nvim.
